src/main/java/uk/gov/companieshouse/api/accounts/CompanyAccountsApplication.java

```diff
@@ -58,13 +63,20 @@ public ObjectMapper objectMapper() {
 
     @Override
     public void addInterceptors(final InterceptorRegistry registry) {
-        registry.addInterceptor(transactionInterceptor)
-                .addPathPatterns("/transactions/{transactionId}/**");
+        registry.addInterceptor(openTransactionInterceptor).order(Ordered.HIGHEST_PRECEDENCE)
+            .addPathPatterns("/transactions/{transactionId}/**");
+
+        registry.addInterceptor(closedTransactionInterceptor).order(Ordered.HIGHEST_PRECEDENCE)
+            .addPathPatterns(
+                "/private/transactions/{transactionId}/company-accounts/{companyAccountId}/filings");
+
         registry.addInterceptor(companyAccountInterceptor)
-                .addPathPatterns(
-                        "/transactions/{transactionId}/company-accounts/{companyAccountId}**");
+            .addPathPatterns(
+                "/transactions/{transactionId}/company-accounts/{companyAccountId}/**",
```

Comment:
the last `/` should not be added before `**` i.e. leave this path as 
`"/transactions/{transactionId}/company-accounts/{companyAccountId}**"`

This change will break the GET/PUT/DELETE methods on the `/transactions/{transactionId}/company-accounts/{companyAccountId}` resource which requires the interceptor to run first to get the data from the DB before proceeding

Reply:
it may be worth adding a comment inline here to explain that the trailing `/` before `**` is left out on purpose to avoid this in future

Reply:
By adding the `/**` 
i.e. ``` /transactions/{transactionId}/company-accounts/{companyAccountId}/**```

it will cater for the following urls' request to be re-directed to the CompanyAccountInterceptor: 
- `transactions/111111-111111-111111/company-accounts/{companyAccountId}`
- `transactions/111111-111111-111111/company-accounts/{companyAccountId}/small-full`
- `transactions/111111-111111-111111/company-accounts/{companyAccountId}/small-full/current-period/`

And it will **NOT re-direct** to the Company Account interceptor the following url: 
- `transactions/111111-111111-111111/company-accounts`


On the other hand, by having only the  `**`
    i.e. `/transactions/{transactionId}/company-accounts/{companyAccountId}**`will cater only for: 
- `transactions/111111-111111-111111/company-accounts/{companyAccountId}`

This means, a request to `transactions/111111-111111-111111/company-accounts/{companyAccountId}/small-full` will not go through the company account interceptor. 
Therefore, it will fail when hitting the SmallFullInterceptor as it needs the CompanyAccountEntity to be in the session.